From 09626eade30d9c41ce18fdcc9f566c3aba9ee716 Mon Sep 17 00:00:00 2001
From: Calvin Johnson <calvin.johnson@nxp.com>
Date: Mon, 30 Apr 2018 11:40:01 +0530
Subject: [PATCH 15/22] staging: fsl_ppfe/eth: unregister netdev after
 pfe_phy_exit

rmmod pfe.ko throws below warning:

kernfs: can not remove 'phydev', no directory
------------[ cut here ]------------
WARNING: CPU: 0 PID: 2230 at fs/kernfs/dir.c:1481
kernfs_remove_by_name_ns+0x90/0xa0

This is caused when the unregistered netdev structure is accessed to
disconnect phy.

Resolve the issue by unregistering netdev after disconnecting phy.

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>

[ Srinidhi Rao : Ported this patch to photon linux from
  qoriq-components linux 'linux-4.14-nxp'
  (https://source.codeaurora.org/external/qoriq/qoriq-components/linux)
]

Signed-off-by: srinidhira0 <srinidhir@vmware.com>
---
 drivers/staging/fsl_ppfe/pfe_eth.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_eth.c b/drivers/staging/fsl_ppfe/pfe_eth.c
index 297aa01b695a..265045fcb008 100644
--- a/drivers/staging/fsl_ppfe/pfe_eth.c
+++ b/drivers/staging/fsl_ppfe/pfe_eth.c
@@ -2464,15 +2464,15 @@ static void pfe_eth_exit_one(struct pfe_eth_priv_s *priv)
 {
 	netif_info(priv, probe, priv->ndev, "%s\n", __func__);
 
-	if (!us) {
+	if (!us)
 		pfe_eth_sysfs_exit(priv->ndev);
 
-		unregister_netdev(priv->ndev);
-	}
-
 	if (!(priv->einfo->phy_flags & GEMAC_NO_PHY))
 		pfe_phy_exit(priv->ndev);
 
+	if (!us)
+		unregister_netdev(priv->ndev);
+
 	if (priv->mii_bus)
 		pfe_eth_mdio_exit(priv->mii_bus);
 
-- 
2.14.2

